app/modules/intelligence/tools/web_tools/code_provider_create_pr.py

```
@@ -0,0 +1,348 @@
+class CodeProviderCreatePullRequestTool:
+    """Tool for creating a pull request in a repository."""
+
+    name: str = "Create a new pull request"
+    description: str = """
+    Create a new pull request in a repository.
+    Useful for proposing and collaborating on changes made in a branch.
+    The tool will create a pull request from your specified head branch to the base branch.
+    """
+    args_schema: Type[BaseModel] = CodeProviderCreatePullRequestInput
+
+    gh_token_list: List[str] = []
```

Ruff (0.14.4)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)